docs/contracts.rst

```diff
@@ -1051,13 +1086,13 @@ but they can be used as base contracts::
         function utterance() public returns (bytes32) { return "miaow"; }
     }
 
+Abstract contracts can implement most of a contract, but leave some methods abstract, facilitating patterns like template method and removing code duplication.
```

Comment:
Could you explain what the template method is?
Also, to remove code duplication, it is not required to use an abstract contract, a base contract is enough.

Reply:
Wikipedia does a good job of explaining the template method, we've used wiki sources before so I assume it's ok to reference that instead of try to redefine it? https://en.wikipedia.org/wiki/Template_method_pattern